File: flutter/lib/ui/components/grid_span_calculator.dart
import 'dart:math' as math;

/// Port of `utils/GridSpanCalculator.kt`.
///
/// Calculates how many columns a responsive grid should have based on the
/// available width. The Kotlin uses 165 dp per column, clamped between 2 and 6.
class GridSpanCalculator {
  static const int _minColumns = 2;
  static const int _maxColumns = 6;
  static const double _columnWidthDp = 165;

  static int columnCount(double availableWidthDp) {
    final columns = (availableWidthDp / _columnWidthDp).floor();
    return math.max(_minColumns, math.min(columns, _maxColumns));
  }
}

File: app/src/main/java/org/ole/planet/myplanet/utils/GridSpanCalculator.kt
package org.ole.planet.myplanet.utils

object GridSpanCalculator {
    private const val MIN_COLUMNS = 2
    private const val MAX_COLUMNS = 6
    private const val COLUMN_WIDTH_DP = 165

    fun columnCount(availableWidthDp: Int): Int {
        val columns = availableWidthDp / COLUMN_WIDTH_DP
        return columns.coerceIn(MIN_COLUMNS, MAX_COLUMNS)
    }
}
